kani-driver/src/call_cbmc.rs

```diff
@@ -160,30 +160,39 @@ impl KaniSession {
     pub fn cbmc_check_flags(&self) -> Vec<OsString> {
         let mut args = Vec::new();
 
-        if self.args.checks.memory_safety_on() {
-            args.push("--bounds-check".into());
-            args.push("--pointer-check".into());
+        // We assume that malloc cannot fail
+        args.push("--no-malloc-may-fail".into());
```

Comment:
Should we consider to switch this on at some point ?

Reply:
@celinval @zhassan-aws any ideas on this? I was quite surprised to see that this was necessary as otherwise a lot of tests newly fail. How does safe Rust avoid UB with code like `let x: Box<usize> = Box::new(i); let y = *x;`?